src/library/f3d_engine.h

```diff
@@ -19,14 +20,29 @@ class interactor;
 class engine
 {
 public:
-  enum class WindowTypeEnum
+  struct window_exception : public std::exception
   {
-    WINDOW_NO_RENDER,
-    WINDOW_OFFSCREEN,
-    WINDOW_STANDARD
+    const char* what() const throw() { return "Cannot create window with this engine"; }
+  };
+  struct interactor_exception : public std::exception
+  {
+    const char* what() const throw() { return "Cannot create interactor with this engine"; }
+  };
+
+  //======== Engine Flags =============
+  // engine::CREATE_WINDOW: Create a window to render into.
+  // engine::CREATE_INTERACTOR: Create an interactor to interact with.
+  // engine::WINDOW_OFFSCREEN: Create an offscreen window to render into, need CREATE_WINDOW.
+  using flags_t = int;
+  enum Flags : flags_t
+  {
+    FLAGS_NONE = 0,             // 0000
+    CREATE_WINDOW = 1 << 0,     // 0001
```

Comment:
`CREATE_WINDOW` and `WINDOW_OFFSCREEN` sound weird to me. What about `WINDOW_NATIVE` and `WINDOW_OFFSCREEN`? And maybe just `INTERACTOR` or `INTERACTOR_DEFAULT`?

Reply:
OFFSCREEN is NATIVE, but OFFSCREEN. With your logic, what happens when you set: 

WINDOW_NATIVE | WINDOW_OFFSCREEN ?

CREATE_* is to signify that without these flags, you will get an exception in the getters.

Reply:
Right, I forgot about that, I thought it was a true offscreen.  
In the future, it would be great to have a real offscreen (using OSMesa or EGL), then how would you differentiate native with hidden window, and offscreen?  
Ideally, the hidden state should be set using the window pointer, wdyt? Or maybe calling explicitly `window->show()`?

Reply:
offscreen shoud be set on creation, before the first render, so no.

Here is how I see it:

You either have a window or you dont (getWindow works or not), controlled by the CREATE_WINDOW flag.
You either have an interactor or you dont (getInteractor works or not), controlled by the CREATE_INTERACTOR flag

the window can be of specific type:
- default
 - offscreen
 - egl
 - osmesa
 - external (?)

Controlled  by WINDOW_* flags

The interactor can be of specific type:
- default
- ?

Controlled  by INTERACTOR_* flags

Maybe the flags should be rennamed but I believe that the flag to create a window or not must exist.